controllers/reaper/reaper_controller_test.go

```diff
@@ -570,6 +572,72 @@ func testCreateReaperWithLocalStorageType(t *testing.T, ctx context.Context, k8s
 	assert.Equal(t, "reaper-data", dataVolumeMount.Name)
 }
 
+func testCreateReaperWithHttpAuthEnabled(t *testing.T, ctx context.Context, k8sClient client.Client, testNamespace string) {
+	t.Log("create the Reaper object")
+	r := newReaper(testNamespace)
+	r.Spec.StorageType = reaperapi.StorageTypeLocal
+	r.Spec.StorageConfig = newStorageConfig()
+	r.Spec.HttpManagement.Enabled = true
+
+	err := k8sClient.Create(ctx, r)
+	require.NoError(t, err)
+
+	t.Log("check that the stateful set is created")
+	stsKey := types.NamespacedName{Namespace: testNamespace, Name: reaperName}
+	sts := &appsv1.StatefulSet{}
+
+	require.Eventually(t, func() bool {
+		return k8sClient.Get(ctx, stsKey, sts) == nil
+	}, timeout, interval, "stateful set creation check failed")
+
+	// if the http auth is enabled, reaper controller should prepare the secret where k8s controller will place per-cluster secrets
+	secretKey := types.NamespacedName{Namespace: testNamespace, Name: reaper.GetTruststoresSecretName(r.Name)}
+	truststoresSecret := &corev1.Secret{}
+	require.Eventually(t, func() bool {
+		return k8sClient.Get(ctx, secretKey, truststoresSecret) == nil
+	}, timeout, interval, "truststore secret creation check failed")
+	assert.True(t, truststoresSecret.Data == nil)
+	assert.Equal(t, 0, len(truststoresSecret.Data))
+
+	// In this configuration, we expect Reaper to also have a mount for the http auth secrets
+	assert.Len(t, sts.Spec.Template.Spec.Containers[0].VolumeMounts, 3)
+	confVolumeMount := sts.Spec.Template.Spec.Containers[0].VolumeMounts[0].DeepCopy()
+	assert.Equal(t, "conf", confVolumeMount.Name)
+	dataVolumeMount := sts.Spec.Template.Spec.Containers[0].VolumeMounts[2].DeepCopy()
+	assert.Equal(t, "reaper-data", dataVolumeMount.Name)
+	truststoresVolumeMount := sts.Spec.Template.Spec.Containers[0].VolumeMounts[1].DeepCopy()
+	assert.Equal(t, "management-api-keystores-per-cluster", truststoresVolumeMount.Name)
+
+	// when we delete reaper, the STS and the secret both go away due to the owner reference
+	// however, that does not happen in env tests
+	err = k8sClient.Delete(ctx, r)
+	require.NoError(t, err)
+
+	reaperKey := types.NamespacedName{Namespace: testNamespace, Name: reaperName}
+	assert.Eventually(t, func() bool {
+		err = k8sClient.Get(ctx, reaperKey, r)
+		return errors.IsNotFound(err)
+	}, timeout, interval, "reaper stateful set deletion check failed")
+
+	assert.Eventually(t, func() bool {
+		err = k8sClient.Get(ctx, stsKey, sts)
+		// we'd expect errors.IsNotFound(err) here, except for some reason this is not happening in env tests
+		return err == nil
+	}, timeout, interval, "reaper stateful set deletion check failed")
+
+	assert.Eventually(t, func() bool {
+		err = k8sClient.Get(ctx, secretKey, truststoresSecret)
+		// again, we'd expect errors.IsNotFound(err) ...
```

Comment:
Issue: can you double check this comment in light of cleanups not happening in envtests. And the one above too...

Reply:
I'm not sure what you meant here, but there's no garbage collection in envtests. So no reference deletions happen as there's no such controller running in envtest. So the comment on line 612 is correct.

Reply:
I wasn't talking about 612, I was talking abot 624.

Reply:
I went back and double checked this.

In the test, I go ahead and delete a Reaper object. This object is pointed to from a stateful set and from a secret via owner reference. The two are owned by the Reaper object. 

When writing this test, I did not know what precisely the _no gargage collection in envtests_ means. The Reaper object got deleted, so I was expecting the other things to disapear too - because that's what they do in an actual cluster. However, without the garbage collection, they won't disappear and I have to delete them manually. 

As a step forward, I propse removing the two `assert.Eventually` checks because they are confusing.

Reply:
I've removed the `assert.Eventually`s.